javascript-algorithms-and-data-structures/learn-basic-string-and-array-methods-by-building-a-music-player-app/scripts.js

```diff
@@ -0,0 +1,231 @@
+    document.querySelector(".play").ariaCurrent = audio.title;
+    document.getElementById("player__album-art").innerHTML = `<img src="${song.cover}" alt="song cover art" />`;
+    audio.play();
+    currentIndex = newIndex;
+
+    updatePlayingSongBackground(newIndex);
+
+}
+
+function pauseSong() {
+  audio.pause();
+  currentTime = audio.currentTime;
+  playSVG.style.fill="#fbf9f9";
+}
+
+function togglePlay() {
+  playSong(songs[currentIndex]);
+  updatePlayingSongBackground(currentIndex)
+
+}
+
+function toggleClosePlaylist() {
+    if (songListDiv.style.visibility == "hidden") {
+        songListDiv.style.visibility = "visible";
+        songListDiv.style.display = "flex";
+        closeBtn.setAttribute('aria-expanded', true);
+
+    } else {
+        songListDiv.style.visibility = "hidden";
+        songListDiv.style.display = "none";
+        closeBtn.setAttribute('aria-expanded', false);
+    }
+}
+
+// player controls
+function nextSong() {
+  const nextIndex = (currentIndex + 1) % songs.length;
+  playSong(songs[nextIndex]);
+  currentIndex = nextIndex;
+
+//   keeping keyboard indicator in sync with array keyls/selected indicator.
+  if(document.activeElement.classList.contains('playlist__song')) {
+    const songDivs = document.querySelectorAll(".playlist__song");
+    songDivs[currentIndex].focus();
+  }
+}
+
+function previousSong() {
+  const prevIndex = (currentIndex - 1 + songs.length) % songs.length;
+  playSong(songs[prevIndex]);
+  currentIndex = prevIndex;
+
+//   keeping keyboard indicator in sync with array keyls/selected indicator.
+  if(document.activeElement.classList.contains('playlist__song')) {
+      const songDivs = document.querySelectorAll(".playlist__song");
+      songDivs[currentIndex].focus();
+    }
+}
+
+// playlist display
+function clearSongBgs() {
+    document.querySelectorAll(".playlist__song").forEach(song => {
+        song.removeAttribute('aria-current');
+    });
+}
+
+function updatePlayingSongBackground(newIndex) {
+    const songDivs = document.querySelectorAll(".playlist__song");
+    songDivs[newIndex].setAttribute('aria-current', true);
+}
+
+function playSelectedSong(song, songBtn) {
+    playSong(song);
+    clearSongBgs();
+    songBtn.setAttribute('aria-current', true);
+}
+
+function shuffle() {
+    // Shallow copy done with spread operator
+    let songsCopy = [...songs];
+    // using array sort method and Math.random method
+    songsCopy.sort(() => 0.5 - Math.random());
+    renderSongs(songsCopy);
+    // reset eventlisteners to stay in sync with songs playing
+    setSongEventListener();
+}
+
+function deleteSong(song){
+  let resetBtn = document.createElement('button');
+  let resetText = document.createTextNode('Reset Playlist');
+  resetBtn.appendChild(resetText);
+  resetBtn.ariaLabel = 'reset playlist';
+  resetBtn.classList.add('player__popup-menu');
+  resetBtn.onclick = function () {
+    app()
+  };
+
+  songs.splice(song,1);
```

Comment:
I also believe this is the first time where splice is introduced

```suggestion
  // first time splice is introduced. make sure to provide explanation and example in the steps. 
  songs.splice(song,1);
```